Assets/HoloToolkit-Examples/Utilities/Scripts/SwapVolume.cs

```diff
@@ -0,0 +1,105 @@
+        private GameObject hideThisObject = null;
+
+        [SerializeField]
+        private GameObject spawnThisPrefab = null;
+
+        [SerializeField]
+        private bool updateSolverTargetToClickSource = true;
+
+        private SolverHandler solverHandler;
+        private bool isOn = false;
+        private GameObject spawnedObject;
+
+        private void Start()
+        {
+            spawnedObject = Instantiate(spawnThisPrefab, hideThisObject.transform.position, hideThisObject.transform.rotation);
+            spawnedObject.SetActive(false);
+            solverHandler = spawnedObject.GetComponent<SolverHandler>();
+        }
+
+        public void OnInputClicked(InputClickedEventData eventData)
+        {
+            if (isOn)
+            {
+                if (spawnedObject != null)
+                {
+                    spawnedObject.SetActive(false);
+                }
+                if (hideThisObject != null)
+                {
+                    hideThisObject.SetActive(true);
+                }
+            }
+            else
+            {
+                if (spawnedObject == null)
+                {
+                    return;
+                }
+
+                spawnedObject.SetActive(true);
+
+                if (updateSolverTargetToClickSource)
+                {
+                    InteractionInputSource interactionInputSource = eventData.InputSource as InteractionInputSource;
+
+                    if (interactionInputSource != null)
+                    {
+                        InteractionSourceInfo sourceKind;
+                        if (interactionInputSource.TryGetSourceKind(eventData.SourceId, out sourceKind))
+                        {
+                            switch (sourceKind)
+                            {
+                                case InteractionSourceInfo.Controller:
+                                    Handedness handedness;
+                                    if (interactionInputSource.TryGetHandedness(eventData.SourceId, out handedness))
+                                    {
+                                        if (handedness == Handedness.Right)
+                                        {
+                                            solverHandler.TrackedObjectToReference = SolverHandler.TrackedObjectToReferenceEnum.MotionControllerRight;
+                                        }
+                                        else
+                                        {
+                                            solverHandler.TrackedObjectToReference = SolverHandler.TrackedObjectToReferenceEnum.MotionControllerLeft;
+                                        }
+                                    }
+                                    break;
+                                default:
+                                    Debug.LogError("The click event came from a device that isn't tracked. Nothing to attach to! Use a controller to select an example.");
```

Comment:
are hands not supported? shouldn't this be supported on HoloLens?

Reply:
I don't believe the Solver system has support for hands. It attaches specifically to the motion controller model.

Reply:
works for me

Reply:
In vNext, solvers should work regardless of controller (ex: hands, clicker, etc)